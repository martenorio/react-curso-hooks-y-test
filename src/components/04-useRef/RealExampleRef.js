import React, { useRef, useState } from 'react';
import { MultipleCustomHook } from '../03-examples/MultipleCustomHook';
import '../02-useEffect/effects.css';

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    const handleClick = () =>{
        setShow(!show)
    }

    return (
        <div>
            <h1>Real Example ref</h1>
            <hr />
            {show && <MultipleCustomHook></MultipleCustomHook>}
            <button onClick={ handleClick } className='btn btn-outline-primary'>{ show ? 'Ocultar' : 'Mostrar'} Tarjeta</button>
        </div>
    )
}
