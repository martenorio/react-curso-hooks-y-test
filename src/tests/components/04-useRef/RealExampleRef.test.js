import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe('RealExampleRef Tests', () => { 
    const wrapper = shallow( <RealExampleRef /> );
    test('should show correctly', () => { 
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHook').exists()).toBe(false);
    })
    test('should show MultipleCustomHool component', () => {
        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHook').exists()).toBe(true);
    })
 })