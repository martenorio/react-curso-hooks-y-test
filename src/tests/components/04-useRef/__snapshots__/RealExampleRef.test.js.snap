// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef Tests should show correctly 1`] = `
<div>
  <h1>
    Real Example ref
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Mostrar
     Tarjeta
  </button>
</div>
`;

exports[`RealExampleRef Tests should show successfully 1`] = `
<div>
  <h1>
    Real Example ref
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Mostrar
     Tarjeta
  </button>
</div>
`;
